daemon/containerd/mount.go

```diff
@@ -0,0 +1,54 @@
+package containerd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/containerd/containerd/mount"
+	"github.com/docker/docker/container"
+	"github.com/sirupsen/logrus"
+)
+
+// Mount mounts the container filesystem in a temporary location, use defer imageService.Unmount
+// to unmount the filesystem when calling this
+func (i *ImageService) Mount(ctx context.Context, container *container.Container) error {
+	snapshotter := i.client.SnapshotService(i.snapshotter)
+	mounts, err := snapshotter.Mounts(ctx, container.ID)
+	if err != nil {
+		return err
+	}
+
+	// The temporary location will be under /var/lib/docker/... because
+	// we set the `TMPDIR`
+	tempMountLocation := os.TempDir()
+
+	root, err := os.MkdirTemp(tempDir, fmt.Sprintf("%s_rootfs-mount", container.ID))
```

Comment:
```suggestion
	root, err := os.MkdirTemp("", fmt.Sprintf("%s_rootfs-mount", container.ID))
```

> [If dir is the empty string, MkdirTemp uses the default directory for temporary files, as returned by TempDir.](https://pkg.go.dev/os#MkdirTemp:~:text=If%20dir%20is%20the%20empty%20string%2C%20MkdirTemp%20uses%20the%20default%20directory%20for%20temporary%20files%2C%20as%20returned%20by%20TempDir.)

Reply:
👍

Reply:
oh! LOL was discussing the same thing with @laurazard before the weekend 😂